src/rule/mod.rs

```diff
@@ -128,14 +128,16 @@ impl FilterRule {
                     "StatePolicy {:?} and protocol {:?} are incompatible",
                     state_policy, proto
                 );
-                bail!(ErrorKind::InvalidRuleCombination(msg));
+                Err(Error::InvalidRuleCombination(msg))
```

Comment:
Would it be better to return `Error::InvalidRuleCombination(StatePolicy, Proto)` as opposed to a pre-reformatted string msg?

Reply:
I don't think the end user needs AddrFamily, because it can be incompatible only when they don't match.
I can create a separate variant for `validate_state_policy` with StatePolicy and Proto, if @faern would approve this. And also separate variants for FilterRuleBuilder and RedirectRuleBuilder.